app/src/main/java/nerd/tuxmobil/fahrplan/congress/dataconverters/LecturesExtensions.kt

```diff
@@ -24,24 +22,16 @@ fun List<Lecture>.toDateInfos() = map(Lecture::toDateInfo)
 
 fun List<Lecture>.toLecturesDatabaseModel() = map(Lecture::toLectureDatabaseModel)
 
-fun List<Lecture>.toDayRanges(timeZoneOffset: ZoneOffset): List<DayRange> {
+fun List<Lecture>.toDayRanges(): List<DayRange> {
     val ranges = mutableSetOf<DayRange>()
     forEach {
-        val dayRange = DayRange(it.getDayStartsAt(timeZoneOffset), it.getDayEndsAt(timeZoneOffset))
+        val day = SystemClock(it.date)
+        val dayRange = DayRange(day, day)
```

Comment:
I made the second parameter optional, defaulting to startDay. Also startsAt and endsAt fields can not be set from outside anymore, since they are always deferred from first two arguments.

Reply:
Much nicer. Thanks.